app/sync.js

```diff
@@ -187,13 +66,40 @@ const doAction = (sender, action) => {
   }
   switch (action.actionType) {
     case syncConstants.SYNC_ADD_SITE:
-      sendSyncRecords(sender, writeActions.CREATE, [createSiteData(action.item.toJS())])
+      sendSyncRecords(sender, writeActions.CREATE,
+        [syncUtil.createSiteData(action.item.toJS())])
       break
     case syncConstants.SYNC_UPDATE_SITE:
-      sendSyncRecords(sender, writeActions.UPDATE, [createSiteData(action.item.toJS())])
+      sendSyncRecords(sender, writeActions.UPDATE,
+        [syncUtil.createSiteData(action.item.toJS())])
       break
     case syncConstants.SYNC_REMOVE_SITE:
-      sendSyncRecords(sender, writeActions.DELETE, [createSiteData(action.item.toJS())])
+      sendSyncRecords(sender, writeActions.DELETE,
+        [syncUtil.createSiteData(action.item.toJS())])
+      break
+    case syncConstants.SYNC_CLEAR_HISTORY:
+      sender.send(messages.DELETE_SYNC_CATEGORY, categoryMap.historySite)
+      break
+    case syncConstants.SYNC_ADD_SITE_SETTING:
+      if (syncUtil.isSyncable('siteSetting', action.item)) {
+        sendSyncRecords(sender, writeActions.CREATE,
+          [syncUtil.createSiteSettingsData(action.hostPattern, action.item.toJS())])
+      }
+      break
+    case syncConstants.SYNC_UPDATE_SITE_SETTING:
+      if (syncUtil.isSyncable('siteSetting', action.item)) {
+        sendSyncRecords(sender, writeActions.UPDATE,
+          [syncUtil.createSiteSettingsData(action.hostPattern, action.item.toJS())])
+      }
+      break
+    case syncConstants.SYNC_REMOVE_SITE_SETTING:
+      sendSyncRecords(sender, writeActions.DELETE,
+        [syncUtil.createSiteSettingsData(action.hostPattern, action.item.toJS())])
+      break
+    case syncConstants.SYNC_CLEAR_SITE_SETTINGS:
+      sender.send(messages.DELETE_SYNC_PREFERENCE_TYPE, 'siteSetting')
+      // TODO: sync-client should listen for this message and delete
+      // all existing preferences with the specified objectData type
```

Comment:
:+1:

`DELETE_SYNC_PREFERENCE_TYPE`, `'device'` sounds odd. i wonder if it's more intuitive to split it into DELETE_SITE_SETTINGS and DELETE_DEVICE.